equinox_monorepo/neural_cde/neural_cde.py

```diff
@@ -0,0 +1,190 @@
+from collections.abc import Callable
+from enum import Enum
+
+import diffrax as dfx
+import equinox as eqx
+import jax
+import jax.nn as jnn
+import jax.numpy as jnp
+import jax.random as jr
+import jax.tree_util as jtu
+from jaxtyping import Array, Float, PRNGKeyArray, PyTree, Scalar
+from typing_extensions import TypeAlias
+
+# These type aliases are unnecessary, but I (packquickly) like them :)
+Linear: TypeAlias = eqx.nn.Linear
+MLP: TypeAlias = eqx.nn.MLP
+Interpolator: TypeAlias = dfx.AbstractGlobalInterpolation
+Solver: TypeAlias = dfx.AbstractSolver
+
+
+class Mode(Enum):
+    terminal = 1
+    meanpool = 2
+
+
+class VectorField(eqx.Module):
+    """The vector field of the neural CDE, a basic feedforward network."""
+
+    mlp: MLP
+    data_size: int
+    state_size: int
+
+    def __init__(
+        self,
+        data_size: int,
+        model_size: int,
+        hidden_width: int,
+        depth: int,
+        activation: Callable,
+        final_activation: Callable,
+        *,
+        key: PRNGKeyArray,
+        **kwargs,
+    ):
+        """**Arguments:**
+
+
+        - `data_size`: The dimension of the input features.
+        - `model_size`: The hidden state size of the model, the dimension of the
+            feedforward network output.
+        - `hidden_width`: The hidden size of the feedforward network, including the
+            output layer.
+        - `depth`: The depth of the feedforward network.
+        - `activation`: The activation between each hidden layer.
+        - `final_activation`: The activation function after the output layer.
+        - `key`: A `jax.random.PRNGKey` used to provide randomness for parameter
+            initialisation.
+        """
+        super().__init__(**kwargs)
```

Comment:
I think we can remove the `**kwargs` and `super().__init__(**kwargs)` as I don't think we will be passing anything to eqx.Module